### ISSUE TITLE
SVG export truncates text in boxes

### ISSUE THREAD

**Issue by Convincible** [jgraph/drawio-desktop#511]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio-desktop/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
When exporting to SVG, any text in boxes does not wrap and is truncated at the point where the first line of text would overflow the box. 

**To Reproduce**
1. Create diagram with boxes containing text that wraps
2. Export to SVG
3. Open SVG e.g. in Illustrator
4. Text is truncated to e.g. "Process step 1 of..."

**Expected behavior**
The full text should be exported into the SVG even if it means that it would overflow the box. That can be corrected in a further step by the user in their SVG editor of choice.

**Screenshots**
![image](https://user-images.githubusercontent.com/30758921/109282459-748d5d80-7815-11eb-8725-4a8e73669e50.png)

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 14.4.3

**Desktop (please complete the following information):**
 - OS: Win10 x64

**Comment by mararad** [comment]
Hi,

could you attach a file so I could test it, please?

Regards,

**Comment by Convincible** [comment]
Sure, can you give an email I can send it to?

**Comment by davidjgraph** [comment]
support@diagrams.net

**Comment by Convincible** [comment]
Thanks @mararad , any chance of a fix at any point for desktop?

**Comment by Convincible** [comment]
Hi @mararad, in Desktop, I export to SVG, and then open the resulting SVG file in Illustrator directly without any intermediate step.

**Comment by mararad** [comment]
Hi,

the only suggestion would be to follow the steps described here: https://desk.draw.io/support/solutions/articles/16000042487-why-text-in-exported-svg-images-may-not-display-correctly i.e untick mentioned fields and style label text. You might want to try to divide them into two lines (where needed) using Enter in order to show labels completely etc. and check if this helps.

SVG won't show labels correctly in IE and most SVG editors (such as Illustrator) but the issue is very complex and so far we haven't find a way to fix this.

Regards,

**Comment by Convincible** [comment]
Hi, yes, the workaround for now is to turn off word wrap before exporting.

Please can I suggest that the programme does this automatically under the hood whenever exporting to SVG? Because there is no occasion where I would want my labels truncated in an export, so I will just have to do it manually every time.

**Comment by raindropsfromsky** [comment]
IMHO the problem is with the viewer, not the image.

I did the same experiment with two shapes. In one, I wrapped the text. In the other, I inserted a manual line break (ENTER) to fold the line. Both have identical behavior when exported to svg.

Instead, the pattern I saw is as follows:
1. If the app is compatible with svg 1.1 standard, the svg image appears perfect.
2. If the app is not compatible with SVG 1.1 standard, then the line is not wrapped in both shapes. <br> In addition, a small line appears at the bottom of the image: "Viewer does not support full SVG 1.1"

The problem happens if you open the svg image in GIMP, or insert it in a document using LibreOffice Writer.

But if you view the svg image in a modern browser (Chrome, Brave, Edge...) the image appears perfect.
The lines are properly wrapped and the warning label does not appear at the bottom.

InkScape is supposed to handle svg properly. But this app also shows one of the two problems: When an svg image is imported in InkScape, it does not show the warning label, but at the same time, the text inside the shapes loses line-wrap.

To sum up, the exported image is OK, but its display varies, depending on the compatibility of the viewer with SVG 1.1 standard.

**Comment by Convincible** [comment]
@raindropsfromsky This isn't my experience. If I export text with word-wrap, and inspect the actual code of the SVG file, the textual information is hard-coded to be truncated. Therefore it doesn't matter what app you open it in.

**Comment by raindropsfromsky** [comment]
I created a simple file, in which there are two shapes. One has natural text-wrap, and the other has hard-coded line-breaks.
[Breaking of text-wrap in exported SVG.zip](https://github.com/jgraph/drawio-desktop/files/6481777/Breaking.of.text-wrap.in.exported.SVG.zip)

I exported the image as SVG.
- If I open the SVG image in a browser, it appears exactly as seen in Draw.io
- If I view the SVG image in FreeCommander's Quickviewer, the text is truncated in both shapes. <br> In addition, a label is attached to the bottom of the image, stating that _the viewer does not support full SVG 1.1_

**Here are the screenshots:**

SVG image as created in Draw.io:
![image](https://user-images.githubusercontent.com/9047168/118345551-e6db2780-b552-11eb-9442-4ae786d8db1e.png)

SVG image as seen in **Brave** browser:
![image](https://user-images.githubusercontent.com/9047168/118345796-86e58080-b554-11eb-9cac-877a46137e53.png)

Image as seen in FreeCommander Quickviewer, GIMP or when inserted in a text document created with Libreoffice Writer:
![image](https://user-images.githubusercontent.com/9047168/118345538-d2972a80-b552-11eb-9f58-e1fb0861b779.png)

You can cross-check the above observations using the enclosed file.

**Comment by davidjgraph** [comment]
23.0.2 now replaces the warning text with a high-res image of the label, if embed images and embed fonts are ticked in the SVG export dialog.

**Comment by maxwxyz** [comment]
This is still an issue in the latest version, please reopen the issue.
I tried automatic wrap, no wrap and manually inserted line spaces but the export always looks truncated and I cannot get all text to be displayed. I tried to open the .svg in Inkscape 1.4, GIMP 3, MS Office, latest LibreOffice Draw 25.2 and Firefox. It looks everywhere the same, just truncated text.

**Comment by davidjgraph** [comment]
Please open a new issue, the original was deemed fixed at the time.

**Comment by cor3ntino** [comment]
Anyone got a solution for that ? It still happens in `"react-drawio": "^1.0.5"`
Thank you

**Comment by davidjgraph** [comment]
If you have a problem that you can recreate on the latest build of this repo, please open a new issue.
